.circleci/config.yml

```diff
@@ -237,6 +260,48 @@ commands:
             gsutil cp  ${TAR_PATH} "gs://${BUILDS_BUCKET}/${ANGULAR_PROJECT_NAME}/${TAR_NAME}"
           working_directory: ~/repo
 
+  conditionally-launch-build-and-deploy:
+    description: Launch build and deploy if project has changed
+    parameters:
+      study_keys:
+        type: string
+      study_guids:
+        type: string
+    steps:
+      - run:
+          name: Check << parameters.study_keys >> and launch build if modified
+          command: |
+            set -u
+            set +e
+            set +x
+            # CircleCi does not support arrays as params, so here is my workaround
+            STUDY_KEYS=(<< parameters.study_keys >>)
+            STUDY_GUIDS=(<< parameters.study_guids >>)
+
+            CHANGED_PROJECTS=$(findmodifiedprojects.sh <<pipeline.git.base_revision >> << pipeline.git.revision >>)
+
+            for i in "${!STUDY_KEYS[@]}"; do
+              STUDY_KEY=${STUDY_KEYS[$i]}
+              STUDY_GUID=${STUDY_GUIDS[$i]}
+
+              NG_PROJECT_NAME="ddp-$STUDY_KEY"
+              echo  $CHANGED_PROJECTS |
+                  grep -E -e $NG_PROJECT_NAME -e _SHARED_  -e ddp-sdk -e toolkit
+              if [[ $? -eq 0 ]]; then
+                echo "Launching build and deploy for $NG_PROJECT_NAME"
+                curl -u "${CIRCLE_API_TOKEN}:" -X POST --header "Content-Type: application/json" -d "{
+                                  \"branch\": \"<< pipeline.git.branch >>\",
+                                  \"parameters\": {
+                                      \"study_key\": \"$STUDY_KEY\",
+                                      \"study_guid\": \"$STUDY_GUID\",
+                                      \"do-builds\": true
+                                  }
+                                }" https://circleci.com/api/v2/project/gh/broadinstitute/ddp-angular/pipeline
+              else
+                echo "Skipping $STUDY_KEY"
+              fi
+            done;
+
```

Comment:
Instead of building all the projects on each git push we are now figuring out what changed from previous build.